sdk/js/src/service/devices/index.js

```diff
@@ -486,14 +488,63 @@ class Devices {
       throw new Error('Missing application ID for device')
     }
 
-    const { supports_join = false, ids = {} } = device
+    const { authenticated_identifiers, target_device_id, ...submitValues } = device
+    const { supports_join = false, ids = {} } = submitValues
 
-    const deviceId = ids.device_id
+    // Initiate claiming, if the device is claimable.
+    const hasAuthenticatedIdentifiers = Boolean(authenticated_identifiers)
+    const claimInfoResponse = await this._api.EndDeviceClaimingServer.GetInfoByJoinEUI({
+      join_eui: ids.join_eui,
+    })
+    const claim = Marshaler.payloadSingleResponse(claimInfoResponse)
+    const supportsClaiming = claim?.supports_claiming ?? false
+    let claimDeviceIds
+    if (supportsClaiming) {
+      // Since this device is claimable, the creation on the join server needs to be skipped.
+      submitValues.join_server_address = undefined
+      const claimPayload = hasAuthenticatedIdentifiers
+        ? {
+            authenticated_identifiers,
+            target_device_id,
+            target_application_ids: {
+              application_id: applicationId,
+            },
+          }
+        : {
+            authenticated_identifiers: {
+              dev_eui: ids.dev_eui,
+              authentication_code: device.claim_authentication_code?.value,
+              join_eui: ids.join_eui,
+            },
+            target_device_id: ids.device_id,
+            target_application_ids: {
+              application_id: applicationId,
+            },
+          }
+      try {
+        const claimResponse = await this._api.EndDeviceClaimingServer.Claim(undefined, claimPayload)
+        claimDeviceIds = Marshaler.payloadSingleResponse(claimResponse)
+      } catch (error) {
+        throw error
+      }
```

Comment:
```suggestion
      } catch (error) {
        throw new Error('Unable to claim end device',  { cause: error })
      }
```

Reply:
You probably just forgot to change this but just to be sure: rethrowing a caught error without modification is a no-op.

Reply:
indeed, I forgot